dev/tools/controllerbuilder/pkg/codebot/tool_run_cmd.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codebot
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/dev/tools/controllerbuilder/pkg/llm"
+	"k8s.io/klog/v2"
+)
+
+func init() {
+	RegisterTool(&RunTerminalCommand{})
+}
+
+type RunTerminalCommand struct {
+	Command string `json:"command"`
+	Args    string `json:"args"`
+}
+
+type RunTerminalCommandResults struct {
+	Success bool   `json:"success"`
+	Output  string `json:"output"`
+	Error   string `json:"error"`
+}
+
+func (t *RunTerminalCommand) Run(ctx context.Context, c *Chat, args map[string]any) (*RunTerminalCommandResults, error) {
+	b, err := json.Marshal(args)
+	if err != nil {
+		return nil, fmt.Errorf("converting to json: %w", err)
+	}
+	if err := json.Unmarshal(b, t); err != nil {
+		return nil, fmt.Errorf("unmarshalling %T: %w", t, err)
+	}
+
+	klog.V(2).Infof("%T: %+v", t, t)
+
+	result := &RunTerminalCommandResults{}
+
+	klog.V(2).Infof("RunTerminalCommand: %+v", t)
+
+	var cmd *exec.Cmd
+
+	tokens := strings.Split(t.Command, " ")
+	switch tokens[0] {
+	case "go":
+		cmd = exec.CommandContext(ctx, "go", tokens[1:]...)
+	case "make":
+		cmd = exec.CommandContext(ctx, "make", tokens[1:]...)
+	default:
+		cmd = exec.CommandContext(ctx, "bash", "-c", t.Command)
+	}
+	cmd.Dir = c.baseDir
+	cmd.Args = append(tokens, strings.Split(t.Args, " ")...)
+
+	stdout, stderr := cmd.CombinedOutput()
+	if stderr != nil {
+		result.Success = false
+		result.Error = stderr.Error()
+	}
```

Comment:
This probably isn't what you mean, it always falls through

Reply:
Ah! You are right. I messed it with the other error when tracing how this tool function is called.